Iwaid 11 crud medicamentos

Mau03gm opened the pull request:
Ticket:
[IWAID-11-CRUD-MEDICAMENTOS](
https://iwaids.atlassian.net/browse/IWAID-11?atlOrigin=eyJpIjoiZjQ4NTM0NzJkZDQxNDg0ZWFhZjcyYjg0YzBiNzNkOGQiLCJwIjoiaiJ9
)

Overview:
Change the table structure and object search, 
create a modal for deletion, update the add and update modals, 
and display dummy objects. 
Modify the form validations to prevent the entry of erroneous data during the data entry process.

How To Test:

Clone the repository
Install the dependencies
In the project directory, you can run:
npm start
This runs the app in the development mode.
Open http://localhost:3000/ to view it in your browser.


Evidence:
![1fbdcead-1ef6-4129-83ca-943d9a3fac11](https://github.com/Chris-THC/iwaid-ui/assets/139919533/868cb420-34b7-44f6-bf51-8c1fa2e47c62)
![cc8feda3-1334-4eec-b748-9b7bdc9cffe4](https://github.com/Chris-THC/iwaid-ui/assets/139919533/63324187-c282-4dd2-b713-0ad186dfcd47)
![cf96ac09-2c7d-4fb1-8f21-57472b47f04d](https://github.com/Chris-THC/iwaid-ui/assets/139919533/61c1224f-d4f6-412f-849b-eda29ae02a6c)
![3e28e87e-f31a-488f-97ef-66eacb6ed41b](https://github.com/Chris-THC/iwaid-ui/assets/139919533/eeadbfa7-ca6e-4044-8a28-dfb2a05f6977)
![d4412d2c-4eed-46b7-b359-de13a86fc96e](https://github.com/Chris-THC/iwaid-ui/assets/139919533/ca4d13e3-330d-4e0f-a899-59f97fd937b6)
![56822215-912b-401b-9f98-24daa1ad1d56](https://github.com/Chris-THC/iwaid-ui/assets/139919533/0a1a7320-5326-4365-9fb5-ade84aefddc5)

Mau03gm committed:
subir crud medicamentos con todas las funcionalidades del crud medicos

Mau03gm committed:
Change the table structure and user search

Mau03gm committed:
Changed, added validations to the form, and correctly placed the titles in Spanish within the form.

Mau03gm committed:
correctly changed the folder name and removed the lines and components that are not in use..

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+      <div>
+      <form onSubmit={handleSubmit(onSubmitClick)}>
+        <div className="form-group">
+          <label>Nombre de la medicina</label>
```
```suggestion
          <label>Nombre del medicamento</label>
```

SendyCruz replied:
```suggestion
          <label>Nombre del medicamento</label>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+          <input
+            type="text"
+            className="form-control"
+            placeholder="Nombre de la medicina"
```
```suggestion
            placeholder="Nombre del medicamento"
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+            defaultValue={isGetData.name}
+          />
+          {errors.name && (
+            <span className="text-danger">El dato es requerido</span>
```
```suggestion
            <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+    step="any" 
+    {...register("dosis", {
+      required: true,
+      min: 0, // Puedes ajustar el mínimo de la dosis según tus necesidades
```
remover comentario

SendyCruz replied:
es correcto que el minimo sea 0? por que?

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+<div className="form-group col-md-4">
+              <label>Presentación</label>
+              <select
+                defaultValue={isGetData.presentation} // Aquí se asigna el valor al select
```
eliminar comentario

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+                {...register("presentation", { required: true })}
+              >
+                <option value="">Seleccione una opción</option>
+                <option value="Sólidos">Sólidos</option>
```
Estos datos así estan definidos en el back? en plural?

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+                <option value="Otro">Otro</option>
+              </select>
+              {errors.presentation && (
+    <span className="text-danger">El dato es requerido</span>
```
```suggestion
    <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,127 @@
+            defaultValue={isGetData.description}
+          ></textarea>
+          {errors.description && (
+            <span className="text-danger">El dato es requerido</span>
```
ditto

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,215 @@
+export function TablaGeneric({ title, data,  }) {
+
+  const [showModalDelete, setShowModalDelete] = useState(false);
+  // const[id, setid] = useState(null);
```
eliminar comentarios

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,215 @@
+  };
+
+  const filteredData = data.filter((item) => {
+    const nombreMatches = item.name
```
cambiar nombre,  usar inglés

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,215 @@
+      .toLowerCase()
+      .includes(searchName.toLowerCase());
+
+    const dosisMatches = item.dosis.includes(searchDosis);
```
ditto

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,215 @@
+
+  const handleClear = () => {
+    setSearchName("");
+    setSearchDosis("");
```
```suggestion
    setSearchDoses("");
```

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,215 @@
+                        className="form-control"
+                        value={searchName}
+                        onChange={(e) => {
+                          setSearchName(e.target.value);
```
usar inglés, aplica para los de abajo tambien

Mau03gm committed:
Comments made:

I correctly changed the folder name and removed the lines and components that are not in use.
Form validations and required data were changed

Mau03gm committed:
remove the lines that are not in use

Mau03gm committed:
removed the lines that are not in use

Chris-THC committed:
Now you can get all medicines on the table

Chris-THC committed:
Merge branch 'iwaid-11-crud-medicamentos' of https://github.com/Chris-THC/iwaid-ui into iwaid-11-crud-medicamentos

cgarcia-iwa commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,253 @@
+                  <th>Dosis</th>
+                  <th>Presentación</th>
+                  <th>Descripción</th>
+                  <th> Cantidad</th>
```
```suggestion
                  <th>Cantidad</th>
```

cgarcia-iwa commented on src/Context/MedicineData.js:
```diff
@@ -0,0 +1,64 @@
+  }
+};
+
+export const createMedicineFunction = async (arrayData) => {
```
estos cambios por que estan en este PR?

cgarcia-iwa commented on src/Context/MedicineData.js:
```diff
@@ -0,0 +1,64 @@
+  // try {
+  //   const response = await axios.post(MedicineURL, arrayData);
+  //   console.log(response.data);
+  console.log(arrayData);
```
borrar los logs

cgarcia-iwa commented on src/Context/MedicineData.js:
```diff
@@ -0,0 +1,64 @@
+
+const MedicineURL = "http://localhost:8081/iwaid/medicines/";
+
+export const getAllMedicineDataFunction = async (setGetAllMedicine) => {
```
cambiar el setGetallMedicine por un nombre mas coherente

Mau03gm committed:
Logs deleted and syntax made coherent.

Mau03gm committed:
Logs deleted and syntax made coherent.

Mau03gm committed:
dose now is dosageForms

Mau03gm committed:
dose now is dosageForms

Mau03gm committed:
key now is code

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+                      {...register("packaging", { required: true })}
+                    >
+                      <option value="">Seleccione una opción</option>
+                      <option value="Liquida">Liquida</option>
```
```suggestion
                      <option value="Liquida">Líquida</option>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+          defaultValue={isGetData.description}
+        ></textarea>
+        {errors.description && (
+          <span className="text-danger">*Dato requerido</span>
```
```suggestion
          <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+        <span className="text-danger">*Dato requerido</span>
+      )}
+      {errors.name?.type === "maxLength" && (
+          <span className="text-danger">Límite superado</span>
```
cual es el limite? Cambiar texto

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+      </div>
+
+      <div className="form-group mb-3">
+      <label>Cantidad</label>
```
sí es requerido, debeía llevar un * rojo a la derecha, revisar cambios hechos en CRUD medicos

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+        defaultValue={isGetData.quantity}
+      />
+      {errors.quantity && (
+        <span className="text-danger">*Dato requerido</span>
```
```suggestion
        <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+                
+
+                  <div className="form-group mb-3">
+        <label>Descripción
```
si es requerido, debeía llevar un * rojo a la derecha, revisar cambios hechos en CRUD medicos

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+          <span className="text-danger">*Máximo 5 caracteres</span>
+        )}
+        {errors.clave?.type === "pattern" && (
+          <span className="text-danger">*Solo caracteres alfanuméricos</span>
```
```suggestion
          <span className="text-danger">Solo caracteres alfanuméricos</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+          <span className="text-danger">*Dato requerido</span>
+        )}
+        {errors.clave?.type === "maxLength" && (
+          <span className="text-danger">*Máximo 5 caracteres</span>
```
```suggestion
          <span className="text-danger">Máximo 5 caracteres</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,211 @@
+          defaultValue={isGetData.code}
+        />
+        {errors.clave?.type === "required" && (
+          <span className="text-danger">*Dato requerido</span>
```
```suggestion
          <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/MedicineList.jsx:
```diff
@@ -0,0 +1,13 @@
+ 
+  const { dataGetAllMedicine } = useContext(GetTheAppContext);
+  
+   //return <TablaGeneric  title='MEDICAMENTOS' data={dataTestMedicine}  />
```
delete

SendyCruz commented on src/Components/Medicine/modalDelete.jsx:
```diff
@@ -0,0 +1,54 @@
+              setTextAlert(`Se eliminó al usuario con id ${idMedicine}`);
+              handleShowFloatAlter();
+            } catch (error) {
+              console.error("Error al agregar el médico:", error);
```
```suggestion
              console.error("Error al eliminar médico:", error);
```

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,252 @@
+        overlay={<Tooltip id="tooltip-clear">Limpiar</Tooltip>}
+      >
+        <button
+          id="iconoClear"
```
es necesario id?

Mau03gm committed:
Spelling has been corrected, and required fields.

Mau03gm committed:
Spelling has been corrected, and required fields.

Mau03gm committed:
Spelling has been corrected, and required fields.

Mau03gm committed:
packaking now is dosageForms

Mau03gm committed:
add {}

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,213 @@
+          defaultValue={isGetData.name}
+        />
+        {errors.name?.type === "required" && (
+          <span className="text-danger">El dato es requerido</span>
```
```suggestion
          <span className="text-danger">Dato requerido</span>
```

SendyCruz commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,213 @@
+        )}
+        {errors.name?.type === "pattern" && (
+          <span className="text-danger">
+            Solo caracteres alfanuméricos y espacios
```
```suggestion
            Solo caracteres alfanuméricos
```

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,253 @@
+                  <tr key={index}>
+                    {displayedFields.map((field) => (
+                      <td key={field}>
+                        <div id="idTextPatient" className="d-inline">
```
patient?

SendyCruz commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,253 @@
+                            onClick={() => {
+                              setDataUserMedicine(item);
+                            }}
+                            id="btnDeletePatient"
```
```suggestion
                            id="btnDeleteMedicine"
```

SendyCruz commented on src/Css/CssTable.css:
```diff
@@ -0,0 +1,70 @@
+    font-size: 12px;
+  }
+
+  #inputSearchPatient {
```
si los estilos son  dedicados por cada componente entonces los de este se deberian llamar con respecto a medicina y no a paciente

cgarcia-iwa commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,213 @@
+        setTextAlert("Medicamento agregado exitosamente");
+        handleShowFloatAlter();
+      } catch (error) {
+        console.error("Error al agregar el medicamento:", error);
```
seria posible agregar los errores con el setTextAlert con alguna variacion?

cgarcia-iwa commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,253 @@
+        onChange={(e) => setSearchPackaging(e.target.value)}
+      >
+        <option value="">Todas las presentaciones</option>
+        {allPackaging.map((packaging, index) => (
```
en back se maneja un nombre distinto, homologar

cgarcia-iwa reviewed (changes requested):
Quitar todos los patiens de los archivos de medicina

SendyCruz committed:
Merge remote-tracking branch 'origin/develop' into iwaid-11-crud-medicamentos

# Conflicts:
#	src/Context/AppContext.jsx

Chris-THC committed:
Now you can create, read, update and delete a medicine

Chris-THC committed:
Now you can create, read, update and delete a medicine

cgarcia-iwa commented on src/Components/Medicine/FormMedicine.jsx:
```diff
@@ -0,0 +1,215 @@
+      handleCloseModal();
+      const responseCreateMedicine = await createMedicineFunction(data);
+
+      if (responseCreateMedicine.status === 201) {
```
puedes guardar en algun tipo de enum los status para evitar usar los numeros directamente en las comparativas

SendyCruz replied:
tambien podría ser una constante, actualizar en los archivos necesarios

cgarcia-iwa commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,256 @@
+
+    const doseMatches = item.dose.includes(searchDoses);
+
+    const packagingMatches = searchPackaging
```
packaging es correcto? en caso de no serlo renombrar todos los casos pendientes

cgarcia-iwa commented on src/Context/MedicineData.js:
```diff
@@ -0,0 +1,53 @@
+
+export const createMedicineFunction = async (arrayData) => {
+  try {
+    console.log(arrayData);
```
borrar logs

cgarcia-iwa commented on src/Context/MedicineData.js:
```diff
@@ -0,0 +1,53 @@
+  try {
+    console.log(arrayData);
+    const response = await axios.post(MedicineURL, arrayData);
+    console.log(response);
```
ditto

Chris-THC committed:
I attended all the PR comments

cgarcia-iwa commented on src/Components/Medicine/modalDelete.jsx:
```diff
@@ -0,0 +1,60 @@
+          onClick={async () => {
+            setActionButtonModal("Eliminar");
+            const responseModalDelete = await deleteMedicineFunction(
+              idMedicine
```
es correcto el valor o es medicineId?

cgarcia-iwa commented on src/Components/Medicine/BackendVariables.js:
```diff
@@ -0,0 +1,7 @@
+const statusCreated = 201;
```
renombrar nombre de archivo para hacer match con los HTTP status

cgarcia-iwa commented on src/Components/Medicine/table.jsx:
```diff
@@ -0,0 +1,259 @@
+                        }
+                      >
+                        <option value="">Todas las presentaciones</option>
+                        {allPackaging.map((packaging, index) => (
```
buscar en los archivos del PR todos los packaging y renombrar en caso de ser necesario

Chris-THC committed:
I attended all the PR comments
